network/network.go

```diff
@@ -196,13 +203,20 @@ func (c *Collector) Collect(ch chan<- prometheus.Metric) {
 			start := time.Now()
 
 			c.logger.Log("level", "info", "message", "dialing host", "host", host, "scrapeID", c.scrapeID)
-			conn, err := c.dialer.Dial("tcp", host)
-			if err != nil {
-				c.logger.Log("level", "error", "message", "could not dial host", "host", host, "scrapeID", c.scrapeID, "stack", microerror.Stack(err))
+			if c.hostIsInPodList(host, pods) {
+				conn, err := c.dialer.Dial("tcp", host)
+				if err != nil {
+					c.logger.Log("level", "error", "message", "could not dial host", "host", host, "scrapeID", c.scrapeID, "stack", microerror.Stack(err))
+					c.dialErrorCount.WithLabelValues(host).Inc()
+					return
+				}
+				defer conn.Close()
+			} else {
+				c.logger.Log("level", "error", "message", "host does not exist", "host", host, "scrapeID", c.scrapeID, "stack", microerror.Stack(err))
+				// TODO: remove this host from future scrapes - how?
```

Comment:
Do we even want to increase the dial error count here? It basically isn't a dial error anymore but instead an inconsistency between the service response and the actual pods.

Reply:
Yeah good point - removing this